### ISSUE TITLE
Sync users from LDAP not works

### ISSUE THREAD

**Issue by nevlkv** [keycloak/keycloak#19781]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

ldap

### Describe the bug

Any task of sync ldap failed finished with with 404  

### Version

21.0.2

### Expected behavior

When i try manual start sync task executed and users sync to database

### Actual behavior

Any task finished with 404 error

### How to Reproduce?

Configure ldap and try sync

env:
``` 
  JAVA_OPTS_APPEND: -Djgroups.dns.query=keycloak-cluster-headless.io.svc.staging.k8s
  KC_LOG_LEVEL: DEBUG,org.keycloak.storage.ldap:TRACE
  KEYCLOAK_ADMIN: user
  KEYCLOAK_CACHE_STACK: kubernetes
  KEYCLOAK_CACHE_TYPE: ispn
  KEYCLOAK_DATABASE_HOST: cockroachdb
  KEYCLOAK_DATABASE_NAME: keycloak
  KEYCLOAK_DATABASE_PORT: "26257"
  KEYCLOAK_DATABASE_USER: root
  KEYCLOAK_ENABLE_HTTPS: "false"
  KEYCLOAK_ENABLE_STATISTICS: "false"
  KEYCLOAK_HTTP_PORT: "8080"
  KEYCLOAK_LOG_OUTPUT: default
  KEYCLOAK_PRODUCTION: "false"
  KEYCLOAK_PROXY: passthrough
```

extraENV
```
extraEnvVars:
  - name: KC_DB_URL
    value: "jdbc:postgresql://cockroachdb:26257/keycloak?sslcert=%2Fcockroach-certs%2Fclient.root.crt&sslkey=%2Fcockroach-certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=%2Fcockroach-certs%2Fca.crt&sslfactory=org.postgresql.ssl.jdbc4.LibPQFactory&user=root"
  - name: KC_DB_VENDOR
    value: postgres
  - name: KC_STORAGE
    value: jpa
  - name: KC_TRANSACTION_XA_ENABLED
    value: "false"
  - name: KC_FEATURES
    value: "account-api,admin-fine-grained-authz,admin2,impersonation,scripts,token-exchange,web-authn,client-policies,ciba,map-storage,par,declarative-user-profile,client-secret-rotation,step-up-authentication,recovery-codes,update-email,preview"

```

debug logs
```
2023-04-18 12:02:45,059 DEBUG [org.hibernate.engine.internal.TwoPhaseLoad] (executor-thread-12) Attribute (`root`)  - enhanced for lazy-loading? - false
2023-04-18 12:02:45,060 DEBUG [org.hibernate.engine.internal.TwoPhaseLoad] (executor-thread-12) Processing attribute `version` : value = 0
2023-04-18 12:02:45,060 DEBUG [org.hibernate.engine.internal.TwoPhaseLoad] (executor-thread-12) Attribute (`version`)  - enhanced for lazy-loading? - false
2023-04-18 12:02:45,060 DEBUG [org.hibernate.engine.internal.TwoPhaseLoad] (executor-thread-12) Done materializing entity [org.keycloak.models.map.storage.jpa.realm.entity.JpaComponentEntity#fa9ebce6-28b1-40a6-9910-b36d5f7461f0]
2023-04-18 12:02:45,060 DEBUG [org.hibernate.engine.loading.internal.CollectionLoadContext] (executor-thread-12) 1 collections were found in result set for role: org.keycloak.models.map.storage.jpa.realm.entity.JpaRealmEntity.components
2023-04-18 12:02:45,060 DEBUG [org.hibernate.engine.loading.internal.CollectionLoadContext] (executor-thread-12) Collection fully initialized: [org.keycloak.models.map.storage.jpa.realm.entity.JpaRealmEntity.components#ad8d6cd4-57f9-41db-912e-4c219e8ed587]
2023-04-18 12:02:45,060 DEBUG [org.hibernate.engine.loading.internal.CollectionLoadContext] (executor-thread-12) 1 collections initialized for role: org.keycloak.models.map.storage.jpa.realm.entity.JpaRealmEntity.components
2023-04-18 12:02:45,060 DEBUG [org.hibernate.loader.collection.plan.AbstractLoadPlanBasedCollectionInitializer] (executor-thread-12) Done loading collection
2023-04-18 12:02:45,060 DEBUG [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-7) Error response 404: javax.ws.rs.NotFoundException: HTTP 404 Not Found
	at org.keycloak.services.resources.admin.RealmAdminResource.extension(RealmAdminResource.java:519)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.constructLocator(ResourceLocatorInvoker.java:107)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.resolveTargetFromLocator(ResourceLocatorInvoker.java:87)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:148)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:140)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:84)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:71)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:140)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:430)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:408)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:140)
	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)

2023-04-18 12:02:45,062 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-7) JtaTransactionWrapper rollback
2023-04-18 12:02:45,063 FINE  [org.postgresql.jdbc.PgConnection] (executor-thread-7)   setAutoCommit = true
2023-04-18 12:02:45,064 DEBUG [org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl] (executor-thread-7) Initiating JDBC connection release from afterTransaction
2023-04-18 12:02:45,064 DEBUG [org.keycloak.transaction.JtaTransactionWrapper] (executor-thread-12) JtaTransactionWrapper  commit
2023-04-18 12:02:45,064 DEBUG [org.hibernate.event.internal.AbstractFlushingEventListener] (executor-thread-12) Processing flush-time cascades


```

### Anything else?

![изображение](https://user-images.githubusercontent.com/46656585/232770544-bf10629b-a057-4c98-b8dc-061d9cf15c4f.png)

![изображение](https://user-images.githubusercontent.com/46656585/232770794-dffab682-3528-4888-ba6e-453ab297d05f.png)
![изображение](https://user-images.githubusercontent.com/46656585/232771009-439db535-927c-467d-946e-cb9ceae82df1.png)

**Comment by nevlkv** [comment]
same behavour on 20.0.5

**Comment by hmlnarik** [comment]
Thank you for the report. The LDAP (or any other federation) does not work with map storage, it only works with the legacy store. For that reason, the synchronization does not work at this moment. This functionality will be available only after #9742 is implemented.
